addons/stock/models/res_config_settings.py

```diff
@@ -58,6 +58,11 @@ def _onchange_group_stock_multi_locations(self):
     def _onchange_group_stock_production_lot(self):
         if not self.group_stock_production_lot:
             self.group_lot_on_delivery_slip = False
+            if any(product.tracking != 'none' for product in self.env['product.product'].search([])):
+                return {
+                    'warning': {
+                        'title': _('Warning!'),
+                        'message': _("You have product(s) in stock that have lot/serial number (i.e. they have some form of tracking enabled). Switching this setting off may lead to problems.")}}
```

Comment:
Don't forget to include this text in the POT file

```suggestion
                        'message': _("You have product(s) tracked by lot/serial number, switching this setting off may lead to errors")}}
```